### ISSUE TITLE
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'name' of undefined

### ISSUE THREAD

**Issue by ardiadrianadri** [UltimateAngular/aot-loader#10]
**Is this a feature request or a bug report?**
 Bug
**What's gone wrong?**
The AoT compilation ends suddenly. Something happens that stops the process

**What was meant to happen?**
I  have two configuration of webpack; one for JiT and other for AoT with the UltimateAngular/aot-loader. The first one works fine but the AoT fails with the next log:

```
> scaff-context@0.0.1-alpha-1 build:prod /Volumes/Seagate Expansion Drive/proyectos/AngularROR/Components/scaff-context
> rimraf dist && webpack --config config/webpack.prod.js --progress --profile --bail

 10% building modules 3/4 modules 1 active ...omponents/scaff-context/src/vendor.ts(node:4185) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: Cannot read property 'name' of undefined
 69% building modules 504/505 modules 1 active ...omponents/scaff-context/src/vendor.ts%
```

**Other information**
Version of webpack: 2.2.0
Version of TypeScript: 2.0.10
Version of UltimateAngular/aot-loader: 0.1.9

webpack.common.js:
```
var webpack = require('webpack');
var HtmlWebpackPlugin = require('html-webpack-plugin');
var helpers = require('./helpers');
var configApp = require ('./configFile.js');

module.exports = {
  entry: {
    'polyfills': './src/polyfills.ts',
    'vendor': './src/vendor.ts',
    'sytles': './src/styles.ts',
    'app': './src/main.ts'
  },

  resolve: {
    extensions: ['.ts', '.js']
  },
  plugins: [
    // Workaround for angular/angular#11580
    new webpack.ContextReplacementPlugin(
      // The (\\|\/) piece accounts for path separators in *nix and Windows
      /angular(\\|\/)core(\\|\/)(esm(\\|\/)src|src)(\\|\/)linker/,
      helpers.root('./src'), // location of your src
      {} // a map of your routes
    ),

    new webpack.optimize.CommonsChunkPlugin({
      name: ['app', 'vendor', 'polyfills','sytles']
    }),

    new HtmlWebpackPlugin({
      template: 'src/index.ejs',
      component: '<'+configApp.componentTag+'>Loading...</'+configApp.componentTag+'>'
    }),

    new webpack.DefinePlugin({
      ENV: configApp
    })
  ]
};
```
webpack.aot.js
```
var webpack = require('webpack');
var webpackMerge = require('webpack-merge');
var ExtractTextPlugin = require('extract-text-webpack-plugin');
var commonConfig = require('./webpack.common.js');
var helpers = require('./helpers');
var aotLoader = require ('@ultimate/aot-loader');

const ENV = process.env.NODE_ENV = process.env.ENV = 'production';

module.exports = webpackMerge(commonConfig, {
  devtool: 'source-map',

  output: {
    path: helpers.root('dist'),
    publicPath: '/',
    filename: '[name].[hash].js',
    chunkFilename: '[id].[hash].chunk.js'
  },

module: {
    rules: [
      {
        test: /\.ts$/,
        loaders: ['@ultimate/aot-loader']
      },
      {
        test: /\.html$/,
        loaders: ['raw-loader','html-loader']
      },
      {
        test: /\.(png|jpe?g|gif|svg|woff|woff2|ttf|eot|ico)$/,
        loaders: ['raw-loader','file-loader?name=assets/[name].[hash].[ext]']
      },
      {
        test: /\.css$/,
        exclude: helpers.root('src', 'app'),
        loader: 'css-loader?sourceMap'
      },
      {
        test: /\.css$/,
        include: helpers.root('src', 'app'),
        loader: 'raw-loader'
      }
    ]
  },

  plugins: [
    new webpack.NoEmitOnErrorsPlugin(),
    new webpack.optimize.UglifyJsPlugin({ // https://github.com/angular/angular/issues/10618
      mangle: {
        keep_fnames: true
      }
    }),
    new webpack.DefinePlugin({
      'process.env': {
        'ENV': JSON.stringify(ENV)
      }
    }),
    new webpack.LoaderOptionsPlugin({
      htmlLoader: {
        minimize: false // workaround for ng2
      }
    }),
    new aotLoader.AotPlugin({
      tsConfig:'./tsconfig.json',
      entryModule: './src/app/app.module#AppModule'
    })
  ]
});
```
tsconfig.json:
```
{
  "compilerOptions": {
    "target": "es5",
    "module": "es2015",
    "moduleResolution": "node",
    "sourceMap": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "lib": ["es2015", "dom"],
    "noImplicitAny": true,
    "suppressImplicitAnyIndexErrors": true
  },
  "exclude": [
    "node_modules"
  ]
}
```

**Comment by rynclark** [comment]
Will take a look at this tonight

**Comment by chasemgray** [comment]
any progress on this? We're seeing this now.

**Comment by rynclark** [comment]
Have you got an example repo? Can't figure out how to replicate this

**Comment by ardiadrianadri** [comment]
Sorry for the delay. I only have time to work in my personal projects in weekends. 

You can reproduce this issue in the code of this github repository: https://github.com/ardiadrianadri/scaff-context/tree/ultimate-aot

The branch is ultimate-aot. 

The AoT compilation runs with the command "npm run build:prod" 

Thank you for your time guys

**Comment by VictorLeach96** [comment]
I managed to fix this by removing the vendor entry point from my webpack config.

**Comment by ardiadrianadri** [comment]
I can confirm that, if I remove the vendor entry, the AoT compilation ends. However, other thing is not working. I'm getting a bundle of 6,8 MB. I think that it seems too much heavy to be the result of an AoT compilation:

```
Hash: 7b754e68cc523cd749f3
Version: webpack 2.2.0
Time: 6945ms
       Asset       Size  Chunks                    Chunk Names
      app.js     6.8 MB       0  [emitted]  [big]  app
polyfills.js     842 kB       1  [emitted]  [big]  polyfills
   sytles.js    12.9 kB       2  [emitted]         sytles
  index.html  497 bytes          [emitted]
```
Another question to answer is why the compilation stops without to finish if there is a vendor entry

**Comment by Itrulia** [comment]
AoT doesnt inlcude tree shaking. AoT is just compiling the HTML to high performant JavaScript (like JSX). Did you do a `webpack -p`?

**Comment by rynclark** [comment]
Fixed in 0.1.12!
